### PR TITLE
Loki: Bump `dskit` to inherit new backoff configs

### PR DESCRIPTION
**Pull request by DylanGuedes**
**What this PR does / why we need it**:
Bump `dskit` to inherit https://github.com/grafana/dskit/pull/296.
It will allow us to configure custom backoff timeouts.

**Which issue(s) this PR fixes**:
N/A

**Special notes for your reviewer**:

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

**Review by JStickler — APPROVED** [review]
[docs team]  LGTM, one question about spelling.

File: docs/sources/configuration/_index.md [hunk 1]
```diff
@@ -389,10 +389,20 @@ grpc_tls_config:
...
+# Optionally log request headers.
+# CLI flag: -server.log-request-headers
+[log_request_headers: <boolean> | default = false]
+
+# Optionally log requests at info level instead of debug level. Applies to
+# request headers as well if server.log-request-headers is enabled.
 # CLI flag: -server.log-request-at-info-level-enabled
 [log_request_at_info_level_enabled: <boolean> | default = false]
 
+# Comma separated list of headers to exclude from loggin. Only used if
```
Comment (JStickler) [inline docs/sources/configuration/_index.md:401]: Should "loggin" be "logging"?

Comment (DylanGuedes) [reply docs/sources/configuration/_index.md:401]: agreed. But we can't really fix this on Loki :cry: this was autogenerated from a dependency. To fix it we'll have to make the change on dskit and then revendor it.

Comment (JStickler) [reply docs/sources/configuration/_index.md:401]: Ah, ok in that case never mind.

**Review by chaudum — APPROVED** [review]

**Pull request merged**